### ISSUE TITLE
Server crashes intermittently

### ISSUE THREAD

**Issue by castwide** [castwide/solargraph#97]
(Reported in https://github.com/castwide/vscode-solargraph/issues/83)

The `completionItem/resolve` method intermittently crashes the server, apparently due to an access violation in a component of the RDoc library that's not thread-safe. A fix is in progress and will be published in the next patch release.

**Comment by gencer** [comment]
Thank you! I am updating now.

**Comment by gencer** [comment]
@castwide , It did crashed again.

```
[Error - 5:38:07 PM] Connection to server is erroring. Shutting down server.
[Error - 5:38:07 PM] Connection to server is erroring. Shutting down server.
[Error - 5:38:07 PM] Connection to server is erroring. Shutting down server.
[Error - 5:38:07 PM] Connection to server is erroring. Shutting down server.
[Error - 5:38:07 PM] Connection to server is erroring. Shutting down server.
[Error - 5:38:07 PM] Connection to server is erroring. Shutting down server.
```

**Comment by castwide** [comment]
Does it crash more or less immediately, during initialization? If not, can you pinpoint what you're doing when it crashes?

What version of Ruby are you using?

**Comment by castwide** [comment]
Thanks. I switched to Ruby 2.5.3 to see if I can reproduce the error. (I've been using 2.5.1.)

**Comment by castwide** [comment]
Fix released in v0.29.0.

**Comment by castwide** [comment]
Closing for inactivity. If anyone still has issues with crashes, please feel free to open a new issue.
